### DISCUSSION TITLE
Changing the language code format of existing translations from POSIX with underscore to BCP style with hyphen

### DISCUSSION THREAD

**Discussion by sr258** [WeblateOrg/weblate#5914] [Q&A]
Hi,

I inadvertently set up Weblate to create new language files in the POSIX format with underscore (e.g. es_MX). The application requires BXP style with hpyhen (es-mx). There are already lots of translations in the POSIX file format now. Is there a way to change the format of existing translations?

**Comment by sr258** [comment]
I ended up doing it with a bash script that I executed from the vcs directory:

```sh
cd $2
mv language/es_MX.json language/es-mx.json
git add language/es_MX.json
git add language/es-mx.json
git commit -m "Renamed es-mx"
weblate loadpo $1/$2
weblate pushgit $1/$2
cd ..
```

**Comment by nijel** [comment]
See https://docs.weblate.org/en/latest/admin/languages.html#adding-new-translations

**Reply by sr258** [reply]
Maybe I overlooked something, but the docs only explain how to change the language code format for newly created translations, not for ones that already exist.

**Reply by nijel** [reply]
Yes, Weblate does not manipulate with existing files and parses various ways of language codes.

**Comment by ChristianPerner** [comment]
Sorry for disturbing your conversation, but my question is very similar to this and I'm not sure if I should file an issue on this.
In the [FAQ](https://docs.weblate.org/en/latest/faq.html#why-does-weblate-use-language-codes-such-sr-latn-or-zh-hant) it is mentioned that language codes defined in RFC 4646 are used (note: RFC 4646 is obsolete and RFC 5646 is in state 'Best Current Practice since September 2009). But in both of these it is mentioned that subtags are separated by a hyphen(-) and not underscore(_).
Can someone explain me this?
I had the problem, that I created the Serbian Latin translation in Weblate but MSBuild didn't create a satellite assembly for this (same problem for nb_NO). So after digging in the source of MSBuild-targets it showed up that it builds it only for RFC 4646 language tags with hyphen.

**Reply by nijel** [reply]
I've clarified the FAQ in c716bec0e3f68c5a61206af0fa9992c1d6ab34df, this topic is in more detail covered in https://docs.weblate.org/en/latest/admin/languages.html

What file format do you use? Windows RC files default to BCP style which uses dashes.

**Reply by ChristianPerner** [reply]
I use .NET resource files, so in my case the file was generated as Resources.sr_Latn.resx from Weblate. But, as already mentioned, MSBuild uses only BCP style, so this is ignored.
And BTW: Thanks for your fast reply!

**Reply by nijel** [reply]
Okay, I've changed the default for resx as well in c6a2d632b88826ca344d762fb64ce5d8a9687121
